src/lib.rs

```diff
@@ -886,7 +879,14 @@ fn paeth_predictor(a: i16, b: i16, c: i16) -> u8 {
     }
 }
 
-pub fn decode(bytes: &[u8]) -> Result<(PngHeader, Vec<u8>), DecodeError> {
+/// Decodes the provided PNG into RGBA pixels.
+///
+/// The returned [`PngHeader`] contains the image’s size, and other PNG metadata which is not
+/// necessary to make use of the pixels (the returned format is always 8-bit-per-component RGBA).
+///
+/// The returned [`Vec`] contains the pixels, represented as `[r, g, b, a]` arrays.
+/// Its length will be equal to `header.width * header.height`.
```

Comment:
Could you add a hint in these doc comments about using `.into_flattened()`? That's a cool Vec API I didn't even know about! I would have instinctively reached for `.collect()` and then maybe run into trouble with extra allocations.